api/secret.go

```diff
@@ -35,6 +37,268 @@ type Secret struct {
 	WrapInfo *SecretWrapInfo `json:"wrap_info,omitempty"`
 }
 
+// TokenID returns the standardized token ID (token) for the given secret.
+func (s *Secret) TokenID() string {
+	if s == nil {
+		return ""
+	}
+
+	if s.Auth != nil && len(s.Auth.ClientToken) > 0 {
+		return s.Auth.ClientToken
+	}
+
+	if s.Data == nil || s.Data["id"] == nil {
+		return ""
+	}
+
+	id, ok := s.Data["id"].(string)
+	if !ok {
+		return ""
+	}
+
+	return id
+}
+
+// TokenAccessor returns the standardized token accessor for the given secret.
+// If the secret is nil or does not contain an accessor, this returns the empty
+// string.
+func (s *Secret) TokenAccessor() string {
+	if s == nil {
+		return ""
+	}
+
+	if s.Auth != nil && len(s.Auth.Accessor) > 0 {
+		return s.Auth.Accessor
+	}
+
+	if s.Data == nil || s.Data["accessor"] == nil {
+		return ""
+	}
+
+	accessor, ok := s.Data["accessor"].(string)
+	if !ok {
+		return ""
+	}
+
+	return accessor
+}
+
+// TokenMeta returns the standardized token metadata for the given secret.
+// If the secret is nil or does not contain an accessor, this returns the empty
+// string. Metadata is usually modeled as an map[string]interface{}, but token
+// metadata is always a map[string]string. This function handles the coercion.
+func (s *Secret) TokenMeta() map[string]string {
```

Comment:
This seems redundant with TokenMetadata below, Can we just use TokenMetadata and have it autodetect? Currently nothing will ever return both an Auth and Data so it should work in all situations.